tests/cross/src/test/scala/tests/pc/InlineValueSuite.scala

```diff
@@ -213,6 +213,106 @@ class InlineValueSuite extends BaseCodeActionSuite with CommonMtagsEnrichments {
     InlineErrors.notLocal,
   )
 
+  // --- scoping tests ----------
+  checkEdit(
+    "scoping",
+    """|object Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = <<f>> + 1
+       |  }
+       |}""".stripMargin,
+    """|object Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = (4 - Demo.j) + 1
+       |  }
+       |}""".stripMargin,
+  )
+
+  checkEdit(
+    "scoping-class",
+    """|class Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = <<f>> + 1
+       |  }
+       |}""".stripMargin,
+    """|class Demo {
+       |  val j: Int = 5
+       |  val f: Int = 4 - j
+       |
+       |  def m() = {
+       |    val j = 10
+       |    val z = (4 - this.j) + 1
+       |  }
+       |}""".stripMargin,
+  )
+
+  // This can be improved by checking if the scope is changed
+  // an allowing to inline with impilicits otherwise.
+  // Alternatively, we can always allow but still display a warning.
+  checkErorr(
+    "scoping-implicit",
```

Comment:
I wonder if this is always an issue. For example the implicit might be `ExecutionContext` so in most cases we will be ok with inlining. Maybe we could instead show a warning? Or ignore altogether, since we can't guard users against all issues.